sentinelhub/download/session.py

```diff
@@ -126,3 +134,125 @@ def _fetch_token(self, request: DownloadRequest) -> JsonDict:
             return oauth_session.fetch_token(
                 token_url=request.url, client_id=self.config.sh_client_id, client_secret=self.config.sh_client_secret
             )
+
+
+_DEFAULT_SESSION_MEMORY_NAME = "sh-session-token"
+_NULL_MEMORY_VALUE = b"\x00"
+
+
+class SessionSharingThread(Thread):
+    """A thread for sharing a token from `SentinelHubSession` object in a shared memory object that can be accessed by
+    other Python processes during multiprocessing parallelization.
+
+    How to use it:
+
+    .. code-block:: python
+
+        thread = SessionSharingThread(session)
+        thread.start()
+
+        # Run a parallelization process here
+        # Use collect_shared_session() to retrieve the session with other processes
+
+        thread.stop()
```

Comment:
It would be useful. The only problem I see is that because `SessionSharingThread` inherits directly from `Thread` users would expect the same interface. The fact that they would now be able to use a thread object with `with` statement might be a bit confusing.

Reply:
In fact, I'm thinking about renaming `stop` into `join` because that is actually the default method used for stopping a thread.

Reply:
I would leave this as is, might come in handy, but I would create a separate object that uses the `with` interface and is the suggested method of use

Reply:
Hm, I'm not a fan of having yet another object for that. Note that `with` functionality can be simply obtained with:

```Python
try:
    # parallelization code
finally:
    thread.stop()
```

Reply:
Unless we add a new object for it I am against adding this functionality. No more *Swiss-army-knife* objects please

Reply:
Ok, now I renamed to `thread.join` which makes much more sense. I also added try-finally statement into documentation example so that users are aware of this.